rabbitmq/datadog_checks/rabbitmq/rabbitmq.py

```diff
@@ -494,18 +499,23 @@ def get_stats(self, instance, base_url, object_type, max_detailed, filters, limi
             # Post a message on the dogweb stream to warn
             self.alert(base_url, max_detailed, len(data), object_type, custom_tags)
 
-        if len(data) > max_detailed:
-            # Display a warning in the info page
-            msg = (
-                "Too many items to fetch. "
-                "You must choose the {} you are interested in by editing the rabbitmq.yaml configuration file"
-                "or get in touch with Datadog support"
-            ).format(object_type)
-            self.warning(msg)
-
-        for data_line in data[:max_detailed]:
+        data_lines_sent = 0
+        for idx, data_line in enumerate(data):
+            if data_lines_sent >= max_detailed:
+                if idx < (len(data) - 1):
```

Comment:
I don't get why you need to check that. It sounds it will filter out the warning uselessly.

Reply:
Agree with Thomas, even if this is the last run of the loop we should display the warning because the last data_line is ignored.